
# Good VS Bad GWAS

Here we investigate the inflation in test statistics from the bad GWASs. We want to find out 1) if the inflation is constant; 2) if it can be predicted in some way.  

## test statistics ~ LD score

The inflation in GWAS test statistics due to confouding is not always constant, as assumed by LDSC. Below we plotted the chi-square test statistics from the bad (colored in red), good (orange), and control (gray) GWASs as a function of LD score ( Fig \@ref(fig:good-vs-bad-by-ldscorebin) ). The linear model assumed by LDSC is only appropriate for four traits, namely chron, pulse, quals, and snoring. For these traits, LDSC accounts for the inflation due to confounding (i.e., intercept) and the inflation due to polygenity (i.e., slope). However, LDSC is inadequate for other traits, where the inflation is non-linear. More specifically, for these traits, the test statistics for SNPs with a low LD score are much more inflated than would be expected by LDSC. Thus, although LDSC can detect the inflation in GWAS test statistics due to confounding, it cannot be used to correct the inflation for some traits.

```{r eval=F, echo=F}
# gwas test statistics
library(vroom)

trait="neur" # bmi
good=vroom(paste0("unrelated/gwas-good/",trait,"-linear.summaries"), col_names=T)
bad=vroom(paste0("gwas-mix/",trait,"-linear.summaries"), col_names=T)
control=vroom(paste0("gwas-control/",trait,"-linear.summaries"), col_names=T)
# note: stat = chi-square test statistic
common=intersect(good$Predictor, bad$Predictor)
common=intersect(common, control$Predictor) # 110,2693 in common
m1=match(common,good$Predictor)
m2=match(common,bad$Predictor)
m3=match(common,control$Predictor)
all=data.frame(Predictor=common, 
               good=good$Stat[m1], 
               bad=bad$Stat[m2], 
               control=control$Stat[m3], stringsAsFactors = F)

# plots 
end=round(max(c(all$good, all$bad)),0)+1
start=round(min(c(all$good, all$bad)),0)
#good vs. bad
png(paste0("fig/",trait,"-gwas-stat-good-vs-bad.png"),
    width =40, height = 20, units = "cm", res=600)
par(mfrow=c(1,2))
plot(all$good, all$bad,
     xlab="good-chisq", ylab="bad-chisq", 
     xlim=c(start, end), 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1, pch=21,  bg="darkgray", col="white", lwd=0.5)
#points(good[fs],bad[fs],cex = 1, pch=21, col="white", bg="orange")
abline(0, 1, col="darkgray", lwd=1.5, lty=1)
#abline(h=threshold, col="red", lwd=1.5, lty=3)
#abline(v=threshold, col="red", lwd=1.5, lty=3)

# good vs. control
plot(all$good, all$control,
     xlab="good-chisq", ylab="control-chisq", 
     xlim=c(start, end), 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1, pch=21,  bg="darkgray", col="white", lwd=0.5)
#points(good[fs],bad[fs],cex = 1, pch=21, col="white", bg="orange")
abline(0, 1, col="darkgray", lwd=1.5, lty=1)
dev.off()

```

```{r good-vs-bad, fig.cap='Test statistics of bad and good GWASs', fig.show='hold', out.width='100%', fig.asp=0.75, fig.align='left', echo=F, eval=F}
knitr::include_graphics(c("fig-to-insert/bmi-gwas-stat-good-vs-bad.png",
                          "fig-to-insert/neur-gwas-stat-good-vs-bad.png" ))
```

```{bash eval=F, echo=F}
#::::
# organize data
#::::

# 1. get overlapping SNPs across GWASs
cp gen/snps-control-gwas.use temp #1,103,182
awk '(NR==FNR){a[$1];next}($1 in a){print $1}' temp gen/snps-unrel-inds.use > overlap-control-good-bad.snps
mv overlap-control-good-bad.snps temp
awk '(NR==FNR){a[$1];next}($1 in a){print $2}' temp doug/ukbb.ldsc > overlap-control-good-bad.snps
wc -l overlap-control-good-bad.snps # 1,102,693 SNPs in common

# 2. get ld scores for overlapping snps
dir=ldsc/eur_w_ld_chr
for chrom in {1..22}; do
zcat $dir/$chrom.l2.ldscore.gz | awk 'NR>1 {print $2, $6}' > ldscore
awk '(NR==FNR){a[$1];next}($1 in a){print $0}' overlap-control-good-bad.snps ldscore > temp
if [ $chrom -eq 1 ]
then 
 mv temp overlap-control-good-bad.ldscore
else
 cat overlap-control-good-bad.ldscore temp > temp2
 mv temp2 overlap-control-good-bad.ldscore
fi
echo $chrom
done

# 3. get test statistics of GWASs 
dir1="ldsc/out-control/"
dir2="ldsc/out-good-gwas/"
dir3="ldsc/out-mix-pop/"

require("vroom")
snps=vroom("overlap-control-good-bad.ldscore", col_names=F)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals",
        "reaction","sbp","snoring","hyper")
# c("fcv", "height","imp")
for(i in 1:length(traits)){
  
  trait=traits[i]
  dat1=vroom(paste0(dir1,trait,".sumstats.gz"), col_names=T)
  dat2=vroom(paste0(dir2,trait,".sumstats.gz"), col_names=T)
  dat3=vroom(paste0(dir3,trait,".sumstats.gz"), col_names=T)
  m1=match(snps$X1, dat1$SNP)
  m2=match(snps$X1, dat2$SNP)
  m3=match(snps$X1, dat3$SNP)
  out=data.frame(SNP=snps$X1,
                 ldscore=snps$X2,
                  control_chisq=dat1$Z[m1]^2,
                  good_chisq=dat2$Z[m2]^2,
                  bad_chisq=dat3$Z[m3]^2, stringsAsFactors = F)
  out=out[complete.cases(out),]
  # bin LD scores according to quantiles
  cutoff=quantile(out$ldscore, probs = seq(0, 1, 0.005))
  out$ldsc_bin=cut(out$ldscore, breaks=cutoff, labels=1:(length(cutoff)-1))
  write.table(out,paste0("summary/",trait,"-gwas-test-stats-compare.txt"), 
              col.names=T, row.names=F, quote=F)
}

#::::
# plot by ldscore bin
#::::

require("vroom")
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals",
        "reaction","sbp","snoring","hyper")
# c("fcv", "height","imp")

png(paste0("fig/gwas-stat-by-ldscbin.png"),
    width =40, height = 30, units = "cm", res=1000)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare.txt"), col_names=T)
out=data.frame(ave_ldscore=tapply(dat$ldscore,INDEX=dat$ldsc_bin, mean),
               ave_control_chisq=tapply(dat$control_chisq,INDEX=dat$ldsc_bin, mean),
               ave_good_chisq=tapply(dat$good_chisq,INDEX=dat$ldsc_bin, mean),
               ave_bad_chisq=tapply(dat$bad_chisq,INDEX=dat$ldsc_bin, mean))

end=round(max(c(out[,2], out[,3], out[,4])),0)
start=0
plot(out$ave_ldscore, out$ave_control_chisq,
     xlab="ldscore bin", ylab="mean chisquare", 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="gray", col="white", lwd=0.5)
points(out$ave_ldscore, out$ave_good_chisq,
       cex = 1.5, pch=21, col="white", bg="orange")
points(out$ave_ldscore, out$ave_bad_chisq,
       cex = 1.5, pch=21, col="white", bg="red")

}
dev.off()

#:::
# plot by raw ldscores
#:::
require("vroom")

traits=c("bmi", "neur")

for(i in 1:length(traits)){
  
trait=traits[i]
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare.txt"), col_names=T)

png(paste0("fig/",trait,"-gwas-stat-by-ldscore.png"),
    width =20, height = 20, units = "cm", res=600)

end=round(max(c(dat$control_chisq, dat$good_chisq, dat$bad_chisq)),0)+1
start=0
plot(dat$ldscore, dat$bad_chisq,
     xlab="ldscore", ylab="chisq", 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1, pch=21,  bg="red", col="white", lwd=0.5)
points(dat$ldscore, dat$control_chisq,
       cex = 1, pch=21, col="white", bg="orange")
points(dat$ldscore, dat$good_chisq,
       cex = 1, pch=21, col="green", bg="red")

dev.off()
}

```

```{r good-vs-bad-by-ldscorebin, fig.cap='Test statistics by LD score for good, bad and control GWASs.', fig.show='hold', out.width='100%', fig.asp=0.75, fig.align='left', echo=F}
knitr::include_graphics(c("fig-to-insert/gwas-stat-by-ldscbin.png"))
```

## New models?

1. * $S_{j} = 1 + n_{j}a +  n_{j}\sum_{i}^{m}(r_{ji}^2 \ q_{i}/Q) h^2_{snp} + \epsilon_{j}$
      * under GCTA: $q_{i}=1, Q=\sum_{i}^{m}q_i=m$ 
      * under LDAK-thin: $q_{i}=I_{i}[f_{i}(1-f_{i})]^{0.75}$

1. * $S_{j} = 1 + n_{j}a +  n_{j}\sum_{i}^{k}(r_{ji}^2 \ q_{i}/Q) h^2_{snp} + n_j \sum ()h + \epsilon_{j}$
      * under GCTA: $q_{i}=1, Q=\sum_{i}^{m}q_i=m$ 
      * under LDAK-thin: $q_{i}=I_{i}[f_{i}(1-f_{i})]^{0.75}$


## LD scores based on individual level data

```{bash eval=F}

mkdir ldsc/ldscore-individual-data
mkdir gen/snps-unrel-inds-by-chrom

# divide snp list by chrom
for i in {1..22}; do
awk -v i=$i '{split($0, a , ":")} (a[1]==i) {print $0}' gen/snps-unrel-inds.use > gen/snps-unrel-inds-by-chrom/snps-unrel-inds-$i
done

# compute LD scores

for i in {1..22}; do
echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 80G
#SBATCH -c 2
#SBATCH -t 10:0:0

.././ldsc.py \
      --bfile ../../gen/geno-mix \
      --extract ../../gen/snps-unrel-inds-by-chrom/snps-unrel-inds-$i \
      --keep ../../mix-pop-gwas.id \
      --l2 \
      --ld-wind-cm 1 \
      --out ../ldscore-individual-data/ldsc-mix-$i
"> sh_script/compute-ldsc-$i.sh
done

for i in {1..22}; do
sbatch -A snpher ../sh_script/compute-ldsc-$i.sh
done > ../../job-records/compute-ldsc-mix

# check job completion
file=job-records/compute-ldsc-mix
jobs=`awk '{print $4}' $file`
mkdir $file-tmp
for i in $jobs; do
jobinfo $i | awk -F ":" -v i=$i '$1~/Name/ {print i, $2}' >> $file-tmp/name.tmp 
jobinfo $i | awk -F ":" '$1~/State/ {print$2}' >> $file-tmp/state.tmp
jobinfo $i | awk -F ":" '$1~/Cores/ {print$2}' >> $file-tmp/cores.tmp
jobinfo $i | awk -F ":" '$1~/Used walltime/ {print $2 ":" $3 ":" $4}' >> $file-tmp/time.tmp
jobinfo $i | awk -F ":" '$1~/Max Mem/ {split($2,a,/[(]/ ); print a[1]}' >> $file-tmp/mem.tmp
done
paste $file-tmp/name.tmp \
      $file-tmp/state.tmp \
      $file-tmp/cores.tmp \
      $file-tmp/time.tmp \
      $file-tmp/mem.tmp \
      | awk 'BEGIN{print "ID name state cores time mem"}{print $0}' > $file.out
rm -r $file-tmp

```


```{bash eval=F, echo=F}
#::::
# organize data
#::::

# 1. get overlapping SNPs across GWASs
awk '(NR==FNR){a[$1];next}($2 in a){print $1}' overlap-control-good-bad.snps doug/ukbb.ldsc > overlap-control-good-bad.snps2 
wc -l overlap-control-good-bad.snps2 # 1,102,693 SNPs in common

# 2. get ld scores for overlapping snps
dir=ldsc/ldscore-individual-data
for chrom in {1..22}; do
zcat $dir/ldsc-mix-$chrom.l2.ldscore.gz | awk 'NR>1 {print $2, $4}' > ldscore
awk '(NR==FNR){a[$1];next}($1 in a){print $0}' overlap-control-good-bad.snps2 ldscore > temp
if [ $chrom -eq 1 ]
then 
 mv temp overlap-control-good-bad.ldscore2
else
 cat overlap-control-good-bad.ldscore2 temp > temp2
 mv temp2 overlap-control-good-bad.ldscore2
fi
echo $chrom
done

# covert to rs system
awk '(NR==FNR){a[$1]; b[$1]=$2; next}($1 in a){print b[$1], $2}' doug/ukbb.ldsc overlap-control-good-bad.ldscore2 > overlap-control-good-bad.ldscore-mix 

# 3. get test statistics of GWASs 
dir1="gwas-control/"
dir2="unrelated/gwas-good/"
dir3="gwas-mix/"

require("vroom")
snps=vroom("overlap-control-good-bad.ldscore", col_names=F)
snps2=vroom("overlap-control-good-bad.ldscore-mix", col_names=F)

snps=cbind(snps,snps2[match(snps$X1,snps2$X1),2], stringsAsFactors=F)
names(snps)=c("snp", "ldsc_ref", "ldsc_mix")

traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp",
        "reaction","sbp","snoring","hyper")

for(i in 1:length(traits)){
  
  trait=traits[i]
  dat1=vroom(paste0(dir1,trait,"-linear-rs.summaries"), col_names=T)
  dat2=vroom(paste0(dir2,trait,"-linear-rs.summaries"), col_names=T)
  dat3=vroom(paste0(dir3,trait,"-linear-rs.summaries"), col_names=T)
  
  m1=match(snps$snp, dat1$SNP)
  m2=match(snps$snp, dat2$SNP)
  m3=match(snps$snp, dat3$SNP)
  out=data.frame(SNP=snps$snp,
                 ldsc_ref=snps$ldsc_ref,
                 ldsc_mix=snps$ldsc_mix,
                  control_chisq=dat1$Z[m1]^2,
                  good_chisq=dat2$Z[m2]^2,
                  bad_chisq=dat3$Z[m3]^2, stringsAsFactors = F)
  out=out[complete.cases(out),]
  # bin LD scores according to quantiles
  cutoff1=quantile(out$ldsc_ref, probs = seq(0, 1, 0.005))
  cutoff2=quantile(out$ldsc_mix, probs = seq(0, 1, 0.005))
  out$ldsc_ref_bin=cut(out$ldsc_ref, breaks=cutoff1, labels=1:(length(cutoff1)-1))
  out$ldsc_mix_bin=cut(out$ldsc_mix, breaks=cutoff2, labels=1:(length(cutoff2)-1))
  write.table(out,paste0("summary/",trait,"-gwas-test-stats-compare2.txt"), 
              col.names=T, row.names=F, quote=F)
}

#::::
# plot by ldscore bin
#::::

require("vroom")
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp",
        "reaction","sbp","snoring","hyper")

png(paste0("fig/gwas-stat-by-ldscbin.png"),
    width =50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5))

for(i in 1:length(traits)){
trait=traits[i]
trait
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare2.txt"), col_names=T)
out1=data.frame(ave_ldscore=tapply(dat$ldsc_ref,INDEX=dat$ldsc_ref_bin, mean),
               ave_control_chisq=tapply(dat$control_chisq,INDEX=dat$ldsc_ref_bin, mean),
               ave_good_chisq=tapply(dat$good_chisq,INDEX=dat$ldsc_ref_bin, mean),
               ave_bad_chisq=tapply(dat$bad_chisq,INDEX=dat$ldsc_ref_bin, mean))
               
#out2=data.frame(ave_ldscore=tapply(dat$ldsc_mix,INDEX=dat$ldsc_mix_bin, mean),
#               ave_control_chisq=tapply(dat$control_chisq,INDEX=dat$ldsc_mix_bin, mean),
#               ave_good_chisq=tapply(dat$good_chisq,INDEX=dat$ldsc_mix_bin, mean),
#              ave_bad_chisq=tapply(dat$bad_chisq,INDEX=dat$ldsc_mix_bin, mean))

end=round(max(c(out1[,2], out1[,3], out1[,4])),0)
#end2=round(max(c(out2[,2], out2[,3], out2[,4])),0)
#end=max(end1,end2)
start=0

plot(out1$ave_ldscore, out1$ave_control_chisq,
     xlab="ldscore", ylab="mean chisquare", 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="gray", col="white", lwd=0.5)
points(out1$ave_ldscore, out1$ave_good_chisq,
       cex = 1.5, pch=21, col="white", bg="orange")
points(out1$ave_ldscore, out1$ave_bad_chisq,
       cex = 1.5, pch=21, col="white", bg="red")
}
dev.off()

#:::
# plot by raw ldscores
#:::
require("vroom")

traits=c("bmi", "neur")

for(i in 1:length(traits)){
  
trait=traits[i]
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare.txt"), col_names=T)

png(paste0("fig/",trait,"-gwas-stat-by-ldscore.png"),
    width =20, height = 20, units = "cm", res=600)

end=round(max(c(dat$control_chisq, dat$good_chisq, dat$bad_chisq)),0)+1
start=0
plot(dat$ldscore, dat$bad_chisq,
     xlab="ldscore", ylab="chisq", 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1, pch=21,  bg="red", col="white", lwd=0.5)
points(dat$ldscore, dat$control_chisq,
       cex = 1, pch=21, col="white", bg="orange")
points(dat$ldscore, dat$good_chisq,
       cex = 1, pch=21, col="green", bg="red")

dev.off()
}

```

## Test the new model

Inflation = np^2 (h2-h2local) <=np^2

1. extract heritability estimates from good GWASs
2. n*p2*h2

3. $S_{true}$ vs $\hat{S}_{ldsc}$ vs $\hat{S}_{new}$ 

```{r eval=F}
# extract heritability estimates
h2_gcta=read.table('summary/reml-gcta-inflation-good-gwas.txt', header=T, 
                stringsAsFactors = F)
h2_ldak=read.table('summary/reml-ldak-thin-inflation-good-gwas.txt', header=T, 
                stringsAsFactors = F)

# average r_ij^2
r2_good=read.table("inflation/summary/ave-r2-1k-snps-goodgwas", header=F)
r2_bad=read.table("inflation/summary/ave-r2-1k-snps-badgwas", header=F)
r2_good=mean(r2_good$V1)
r2_bad=mean(r2_bad$V1)

# ldsc intercept estimates
ldsc_bad=read.table("summary/ldsc-mix", header=F, stringsAsFactors = F)

# GWAS test stats & LD scores

require("vroom")
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals",
        "reaction","sbp","snoring","hyper")
# c("fcv", "height","imp")

png(paste0("fig/predcited-chisq-by-ldscbin.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))

for(i in 1:length(traits)){
  
n=100000
trait=traits[i]
ldsc_intercept=ldsc_bad$V2[ldsc_bad$V1==trait]
h2=h2_ldak$all_est[h2_ldak$code==trait]
  
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare2.txt"), col_names=T)
dat$chisq_ldsc=dat$good_chisq+ldsc_intercept-1
dat$chisq_new=dat$good_chisq+n*r2_bad*h2

out=data.frame(ave_ldscore=tapply(dat$ldsc_ref,INDEX=dat$ldsc_ref_bin, mean),
               ave_chisq_ldsc=tapply(dat$chisq_ldsc,INDEX=dat$ldsc_ref_bin, mean),
               ave_chisq_new=tapply(dat$chisq_new,INDEX=dat$ldsc_ref_bin, mean),
               ave_good_chisq=tapply(dat$good_chisq,INDEX=dat$ldsc_ref_bin, mean),
               ave_bad_chisq=tapply(dat$bad_chisq,INDEX=dat$ldsc_ref_bin, mean))

end=round(max(c(out[,2], out[,3], out[,4]), out[,5]),0)
start=round(min(c(out[,2], out[,3], out[,4]), out[,5]),0)

plot(out$ave_ldscore, out$ave_bad_chisq,
     xlab="ldscore bin", ylab="mean chisquare", 
     ylim=c(start, end),
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
points(out$ave_ldscore, out$ave_chisq_new,
       cex = 1.5, pch=21, col="white", bg="orange")
points(out$ave_ldscore, out$ave_chisq_ldsc,
       cex = 1.5, pch=21, col="white", bg="red")
#points(out$ave_ldscore, out$ave_good_chisq,
#       cex = 1.5, pch=21, col="white", bg="gray")
}
dev.off()

```

## predictors of inflation?

Potential predictors considered:
1. MAF
2. ave r^2_ij for each j cross chromsomes [10k SNPs]

```{bash eval=F}

# prep: convert snp labels to rs system
awk '(NR==FNR){a[$1]; b[$1]=$2; next}($1 in a){print b[$1], $2, $3, $5}' doug/ukbb.ldsc gen/geno-unrel.stats > gen/geno-unrel-rs.maf

R
require("vroom")
# files for maf & ave r2 by snp
stat=vroom("gen/geno-unrel-rs.maf", col_names=F)
r2_good=vroom("summary/ave-r2-by-snp-goodgwas-rs", col_names=F)
r2_bad=vroom("summary/ave-r2-by-snp-badgwas-rs", col_names=F)

# compute inflation in chisq (good vs. bad) by trait
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

for(i in 1:length(traits)){
  
trait=traits[i]
dat=vroom(paste0("summary/",trait,"-gwas-test-stats-compare2.txt"), col_names=T)
inflation=data.frame(snp=dat$SNP,
                     inflation=dat$bad_chisq-dat$good_chisq,
                     bad_chisq=dat$bad_chisq,
                     good_chisq=dat$good_chisq,
                     stringsAsFactors = F)

m1=match(inflation$snp, stat$X1)
m2=match(inflation$snp,r2_good$X1)
m3=match(inflation$snp,r2_bad$X1)
out=data.frame(inflation,
               maf=stat$X4[m1], 
               r2_good=r2_good$X2[m2],
               r2_bad=r2_bad$X2[m2], stringsAsFactors = F)

# bin maf & r2_bad according to quantiles
cutoff1=quantile(out$maf, probs = seq(0, 1, 0.005), na.rm=T)
cutoff2=quantile(out$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)

out$maf_bin=cut(out$maf, breaks=cutoff1, labels=1:(length(cutoff1)-1))
out$r2_bad_bin=cut(out$r2_bad, breaks=cutoff2, labels=1:(length(cutoff2)-1))

write.table(out,paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), 
            col.names=T, row.names=F, quote=F)
}

# plot by maf
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
# "fcv", "height","imp",
png(paste0("fig/inflation-by-mafbin.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","maf","maf_bin")]
sel=sel[complete.cases(sel),]
out=data.frame(inflation=tapply(sel$inflation,INDEX=sel$maf_bin, mean))
out$maf_bin=1:dim(out)[1]
out$maf_bin_val=tapply(sel$maf,INDEX=sel$maf_bin, mean)
  
plot(out$maf_bin_val, out$inflation,
     xlab="maf bin", ylab="mean inflation",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()

# plot by r2_bad bin

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
# "fcv", "height","imp",
png(paste0("fig/inflation-by-aver2.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","r2_bad","r2_bad_bin")]
sel=sel[complete.cases(sel),]
out=data.frame(inflation=tapply(sel$inflation,INDEX=sel$r2_bad_bin, mean))
out$r2_bad_bin=1:dim(out)[1]
out$r2_bad_bin_val=tapply(sel$r2_bad,INDEX=sel$r2_bad_bin, mean)
  
plot(out$r2_bad_bin_val, out$inflation,
     xlab="ave r2 bin", ylab="mean inflation",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()

```

```{r eval=F}

# Here we want to include a plot to show that inflation is not constant across the genome. i.e., LDSC is wrong.
# we plot inflation ~ maf for bmi

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
trait=traits[2]

png(paste0("fig/inflation-by-mafbin-bmi.png"),
    width =30, height = 20, units = "cm", res=600)
par(cex.lab=1.2, font.lab=2)

dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","maf","maf_bin")]
sel=sel[complete.cases(sel),]
out=data.frame(inflation=tapply(sel$inflation,INDEX=sel$maf_bin, mean))
out$maf_bin=1:dim(out)[1]
out$maf_bin_val=tapply(sel$maf,INDEX=sel$maf_bin, mean)
  
plot(out$maf_bin_val, out$inflation,
     xlab="maf bin", ylab=" est. inflation (confoudned chisq - good chisq)",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)

dev.off()


```



```{r eval=F}

## inflation by raw ave r2 
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
# "fvc", "height","imp",
png(paste0("fig/inflation-by-aver2-raw.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","r2_bad")]
sel=sel[complete.cases(sel),]
  
plot(sel$r2_bad, sel$inflation,
     xlab="ave r2", ylab="inflation",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()


###
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
# "fcv", "height","imp",
png(paste0("fig/inflation-by-aver2-goodgwas-raw.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","r2_good")]
sel=sel[complete.cases(sel),]
  
plot(sel$r2_good, sel$inflation,
     xlab="ave r2", ylab="inflation",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()


## histogram
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", 
        "reaction","sbp","snoring","hyper")
# "fcv", "height","imp",
png(paste0("fig/hist-aver2-goodgwas.png"),
    width =40, height = 30, units = "cm", res=600)
par(mfrow=c(3,4))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","inflation","r2_good")]
sel=sel[complete.cases(sel),]
  
hist(sel$r2_good, breaks=100, main=trait)
}
dev.off()

```

## regress chisq ~ mean r^2_j

THIS IS NEW

```{bash eval=F}

#::::::::::::::::::::::
# organize data
#::::::::::::::::::::::

# bad gwas-----------------------------------------------------------------------
mkdir gwas-mix-out

for trait in {awake,bmi,chron,ever,fvc,height,imp,neur,pulse,quals,reaction,sbp,snoring,hyper}; do
gwas=gwas-mix/$trait-linear.summaries
aver2=summary/ave-r2-by-snp-badgwas
ldsc=snps-unrel-inds.ldscore
out=gwas-mix-out/$trait.out

awk 'NR>1 {print $1, $5}' $gwas > tmp/gwas.tmp

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $ldsc tmp/gwas.tmp > tmp/ldsc.tmp

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $aver2 tmp/gwas.tmp > tmp/aver2.tmp

# put info together
paste tmp/gwas.tmp \
      tmp/aver2.tmp \
      tmp/ldsc.tmp \
      | awk 'BEGIN{OFS=";" ; 
                  print "snp;chisq;aver2;ldsc"}
             {$1=$1}1' > $out
done

# good gwas-----------------------------------------------------------------------
mkdir unrelated/gwas-good-out

for trait in {awake,bmi,chron,ever,fvc,height,imp,neur,pulse,quals,reaction,sbp,snoring,hyper}; do
gwas=unrelated/gwas-good/$trait-linear.summaries
aver2=summary/ave-r2-by-snp-goodgwas
out=unrelated/gwas-good-out/$trait.out

awk 'NR>1 {print $1, $5}' $gwas > tmp/gwas.tmp 

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $ldsc tmp/gwas.tmp > tmp/ldsc.tmp

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $aver2 tmp/gwas.tmp > tmp/aver2.tmp

# put info together
paste tmp/gwas.tmp \
      tmp/aver2.tmp \
      tmp/ldsc.tmp \
      | awk 'BEGIN{OFS=";" ; 
                  print "snp;chisq;aver2;ldsc"}
             {$1=$1}1' > $out
done

# ::::::::::::::::::::::::::::::::::::::::::::::
# estimate slope: bad chisq ~ aver2_j & correct the test statistics
#::::::::::::::::::::::::::::::::::::::::::::::

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

for(i in 1:length(traits)){

trait=traits[i]

file=paste0("gwas-mix-out/",trait,".out")

dat=vroom(file, col_names=T, delim=";")
dat=dat[complete.cases(dat),]
mod=lm(chisq ~ aver2,data=dat)

# correct inflation
cor=dat$chisq-dat$aver2*coef(mod)[2]
out=cbind(dat,cor)
write.table(out,paste0("gwas-mix-out/",trait,"-corrected.out"), col.names=T,
            row.names=F, quote=F)

# collect slope
slope0=data.frame(trait=trait,
                  slope_aver2=coef(mod)[2],
                  p_aver2=summary(mod)$coefficients[,4][2],
                  stringsAsFactors = F)

if(i==1){slope=slope0} else {slope=rbind(slope,slope0)}

}

# ::::::::::::::::::::::::::::::::::::::::::::::
# corrected chisq VS good chisq vs. control chisq
#::::::::::::::::::::::::::::::::::::::::::::::

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png("fig/good-vs-corrected-chisq.png", res=600, width=50, height=30, units="cm")
par(mfrow=c(3,5), pty="s")
for(i in 1:length(traits)){
trait=traits[i]
good=vroom(paste0("unrelated/gwas-good-out/",trait,".out"), col_names=T)
good=good[complete.cases(good),]
cor=vroom(paste0("gwas-mix-out/",trait,"-corrected.out"), col_names=T)
m=match(good$snp, cor$snp)
dat=data.frame(snp=good$snp,
               good=good$chisq,
               cor=cor$cor[m])
               
start=min(dat$good,dat$cor)
end=max(dat$good,dat$cor)

plot(dat$good, dat$cor,
     xlim=c(start,end), ylim=c(start, end),
     xlab="good chisq", ylab="corrected chisq", 
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(0, 1, col="darkgray", lwd=1.5, lty=1)

}
dev.off()


```

### plots

```{r eval=F}

#:::::::::::::::::::::::::::
# good gwas
#:::::::::::::::::::::::::::

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png("fig/chisq-aver2-bin-good-gwas.png", res=600, width=50, height=30, units="cm")
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5)

for(i in 1:length(traits)){
  trait=traits[i]
  dat=vroom(paste0("unrelated/gwas-good-out/",trait,".out"), col_names=T)
  dat=dat[complete.cases(dat),]

  # linear model
  mod=lm(chisq ~ aver2 + ldsc,data=dat)
  p=summary(mod)$coefficients[,4][2]
  p_sci=formatC(p, format="e", digit=2)

  # bin a variable by quantile
  cutoff=quantile(dat$aver2, probs = seq(0, 1, 0.005), na.rm=T)
  dat$bin=cut(dat$aver2, breaks=cutoff, labels=1:(length(cutoff)-1))
  
  # average chisq by bin values
  out=data.frame(chisq_ave=tapply(dat$chisq,INDEX=dat$bin, mean))
  out$bin_val=tapply(dat$aver2,INDEX=dat$bin, mean)
  px=min(out$bin_val)
  py=max(out$chisq_ave)

  # plot
  #png("fig/chisq-aver2-bin-good-gwas.png", res=600, width=50, height=30, units="cm")
  #par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5)
  plot(out$bin_val, out$chisq_ave,
           xlab="ave r2 bin", ylab="mean chisq",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(lm(chisq ~ aver2,data=dat),col="orange", lwd=2)
  if(p<0.01){text(px, py, paste0("p = ", p_sci),
                  adj=c(0,1), col="red", cex=2)
  } else {text(px, py, paste0("p = ", p_sci), 
          adj=c(0,1), cex=1.2, font=2)}
  #dev.off()
  
}
dev.off()

#:::::::::::::::::::::::::::
# bad gwas
#:::::::::::::::::::::::::::

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png("fig/chisq-aver2-bin-bad-gwas.png", res=600, width=50, height=30, units="cm")
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5)

for(i in 1:length(traits)){
  trait=traits[i]
  dat=vroom(paste0("gwas-mix-out/",trait,".out"), col_names=T)
  dat=dat[complete.cases(dat),]

  # linear model
  mod=lm(chisq ~ aver2 + ldsc,data=dat)
  p=summary(mod)$coefficients[,4][2]
  p_sci=formatC(p, format="e", digit=2)

  # bin a variable by quantile
  cutoff=quantile(dat$aver2, probs = seq(0, 1, 0.005), na.rm=T)
  dat$bin=cut(dat$aver2, breaks=cutoff, labels=1:(length(cutoff)-1))
  
  # average chisq by bin values
  out=data.frame(chisq_ave=tapply(dat$chisq,INDEX=dat$bin, mean))
  out$bin_val=tapply(dat$aver2,INDEX=dat$bin, mean)
  px=min(out$bin_val)
  py=max(out$chisq_ave)

  # plot
  plot(out$bin_val, out$chisq_ave,
           xlab="ave r2 bin", ylab="mean chisq",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(lm(chisq ~ aver2 ,data=dat),col="orange", lwd=2)
  if(p<0.01){text(px, py, paste0("p = ", p_sci),
                  adj=c(0,1), col="red", cex=2)
  } else {text(px, py, paste0("p = ", p_sci), 
          adj=c(0,1), cex=1.2, font=2)}
  
}
dev.off()

```

Below is the OLD code.
```{r eval=F}

#::::::::
# by binned ave r2
#:::::::

# bad chisq by binned ave r2 
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")
png(paste0("fig/badgwas-chisq-by-aver2-bin.png"),
    width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","bad_chisq","r2_bad","r2_bad_bin")]
sel=sel[complete.cases(sel),]
out=data.frame(bad_chisq=tapply(sel$bad_chisq,INDEX=sel$r2_bad_bin, mean))
out$r2_bad_bin=1:dim(out)[1]
out$r2_bad_bin_val=tapply(sel$r2_bad,INDEX=sel$r2_bad_bin, mean)

plot(out$r2_bad_bin_val, out$bad_chisq,
     xlab="ave r2", ylab="chisq test stat",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()

#::::::::
# by raw ave r2
#:::::::

# bad chisq by raw ave r2 
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp",
        "reaction","sbp","snoring","hyper")

png(paste0("fig/badgwas-chisq-by-aver2.png"),
    width =50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","bad_chisq","r2_bad")]
sel=sel[complete.cases(sel),]
  
plot(sel$r2_bad, sel$bad_chisq,
     xlab="ave r2", ylab="chisq test stat",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()

# good chisq by raw ave r2 
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp",
        "reaction","sbp","snoring","hyper")

png(paste0("fig/goodgwas-chisq-by-aver2.png"),
    width =50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5))
for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","good_chisq","r2_good")]
sel=sel[complete.cases(sel),]
  
plot(sel$r2_good, sel$good_chisq,
     xlab="ave r2", ylab="chisq test stat",
     main=trait, las=1,
     cex = 1.5, pch=21,  bg="grey", col="white", lwd=0.5)
}
dev.off()

# regression: chisq test stats ~ ave r^2_j --------------------------------------
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp",
        "reaction","sbp","snoring","hyper")

for(i in 1:length(traits)){
trait=traits[i]
dat=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
sel=dat[,c("snp","bad_chisq","good_chisq","r2_bad", "r2_good", "maf")]
sel=sel[complete.cases(sel),]
sel$maf2=sel$maf^2

mod1=lm(sel$bad_chisq~sel$r2_bad)
#mod1.1=lm(sel$bad_chisq~sel$maf)
#mod1.2=lm(sel$bad_chisq~ sel$maf + sel$maf2)
#mod1.3=lm(sel$bad_chisq~sel$r2_bad+sel$maf)
#mod1.4=lm(sel$bad_chisq~sel$r2_bad+sel$maf+sel$maf2)
mod2=lm(sel$good_chisq~sel$r2_good)

slope0=data.frame(trait=trait,
                  slope_bad=coef(mod1)[2],
                  p_bad=summary(mod1)$coefficients[,4][2],
                  slope_good=coef(mod2)[2],
                  p_good=summary(mod2)$coefficients[,4][2],
                  stringsAsFactors = F)

if(i==1){slope=slope0} else {slope=rbind(slope,slope0)}
}

slope$good_div_n=slope$slope_good/100000
slope$bad_div_n=slope$slope_bad/100000

# heritability estimates
h2_gcta=read.table('summary/reml-gcta-inflation-good-gwas.txt', header=T, 
                stringsAsFactors = F)
h2_ldak=read.table('summary/reml-ldak-thin-inflation-good-gwas.txt', header=T, 
                stringsAsFactors = F)
h2=data.frame(trait=h2_gcta$code,gcta=h2_gcta$all_est, ldak=h2_ldak$all_est)

# together
m=match(slope$trait,h2$trait)
all=cbind(slope,h2[m,-1])

```

## predict inflation | aver2_j ?

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")
trait=traits[2]

# putting info together
file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
dat=vroom(file, col_names=T)
sel=c(1:2,5,6)
r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
m=match(r2$snp, dat$SNP)
use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
use=use[complete.cases(use),]

# inflation
use$inflation=use$bad_chisq-use$good_chisq
good=lm(good_chisq~ldsc_ref, data=use)
good_hat=predict(good)
bad=lm(bad_chisq~r2_bad, data=use)
bad_hat=predict(bad)
inflation_hat=bad_hat-good_hat
cor(inflation_hat,use$inflation) # 0.89
use$inflation_hat=inflation_hat

dat=use

# bin a variable by quantile
cutoff=quantile(dat$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)
dat$bin=cut(dat$r2_bad, breaks=cutoff, labels=1:(length(cutoff)-1))
  
# average chisq by bin values
out=data.frame(inflation_ave=tapply(dat$inflation,INDEX=dat$bin, mean),
               inflation_hat_ave=tapply(dat$inflation_hat,INDEX=dat$bin, mean))
out$bin_val=tapply(dat$r2_bad,INDEX=dat$bin, mean)

# plot: inflation vs. inflation_hat ~ aver2 bin
png("fig/inflation-vs-predicted-aver2-bin-badgwas-bmi.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(out$bin_val, out$inflation_ave,
         xlab="ave r2 bin", ylab="inflation",
         main=trait, las=1,
         cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
points(out$bin_val, out$inflation_hat_ave, cex=1.5, pch=21, col="orange")
dev.off()

# plot: inflation vs. inflation_hat ~ aver2 raw
png("fig/inflation-vs-predicted-aver2-raw-badgwas-bmi.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(dat$r2_bad, dat$inflation,
         xlab="ave r2 ", ylab="inflation",
         main=trait, las=1,
         cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
points(dat$r2_bad, dat$inflation_hat, cex=1.5, pch=21, col="orange")
dev.off()

# plot: inflation vs. inflation_hat: ave
start=min(c(out$inflation_ave,out$inflation_hat_ave))
end=max(c(out$inflation_ave,out$inflation_hat_ave))
    
png("fig/true-vs-predicted-inflation-ave.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(out$inflation_ave, out$inflation_hat_ave,
     xlab="ave inflation", ylab="ave inflation hat",
     main=" bmi", 
     xlim=c(start, end), ylim=c(start, end),
     cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(0, 1, col="orange", lwd=1.5, lty=1)
points(out$inflation_ave, out$inflation_hat_ave,,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
dev.off()

# plot: inflation vs. inflation_hat
start=min(c(dat$inflation,dat$inflation_hat))
end=max(c(dat$inflation,dat$inflation_hat))
    
png("fig/true-vs-predicted-inflation.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(dat$inflation, dat$inflation_hat,
     xlab="ave inflation", ylab="ave inflation hat",
     main=" bmi", 
     xlim=c(start, end), ylim=c(start, end),
     cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(0, 1, col="orange", lwd=1.5, lty=1)
points(dat$inflation, dat$inflation_hat,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
dev.off()

#..............
write.table(use,"summary/bmi-inflation.use", col.names=T, row.names=F, quote=F)

## other things to try & consider
use$inflation=use$bad_chisq-use$good_chisq
mod1=lm(bad_chisq~ldsc_ref, data=use)
mod2=lm(good_chisq~ldsc_ref, data=use)

e1=resid(mod1)
new_chisq=e1+7.778390
mod3=lm(new_chisq ~ use$r2_bad )
mod4=lm(bad_chisq~r2_bad, data=use)
mod5=lm(inflation~r2_bad, data=use)
use$inflation_hat=predict(mod3)

mod6=lm(bad_chisq~r2_bad+ldsc_ref, data=use)

dat=use

# bin a variable by quantile
cutoff=quantile(dat$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)
dat$bin=cut(dat$r2_bad, breaks=cutoff, labels=1:(length(cutoff)-1))
  
# average chisq by bin values
out=data.frame(inflation_ave=tapply(dat$inflation,INDEX=dat$bin, mean),
               inflation_hat_ave=tapply(dat$inflation_hat,INDEX=dat$bin, mean))
out$bin_val=tapply(dat$r2_bad,INDEX=dat$bin, mean)

# plot: inflation vs. inflation_hat ~ aver2 bin
png("fig/inflation-vs-predicted-aver2-bin-badgwas-bmi.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(out$bin_val, out$inflation_ave,
         xlab="ave r2 bin", ylab="inflation",
         main=trait, las=1,
         cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
points(out$bin_val, out$inflation_hat_ave, cex=1.5, pch=21, col="orange")
dev.off()

# plot: inflation vs. inflation_hat ~ aver2 raw
png("fig/inflation-vs-predicted-aver2-raw-badgwas-bmi.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(dat$r2_bad, dat$inflation,
         xlab="ave r2 ", ylab="inflation",
         main=trait, las=1,
         cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
points(dat$r2_bad, dat$inflation_hat, cex=1.5, pch=21, col="orange")
dev.off()

# plot: inflation vs. inflation_hat: ave
start=min(c(out$inflation_ave,out$inflation_hat_ave))
end=max(c(out$inflation_ave,out$inflation_hat_ave))
    
png("fig/true-vs-predicted-inflation-ave.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(out$inflation_ave, out$inflation_hat_ave,
     xlab="ave inflation", ylab="ave inflation hat",
     main=" bmi", 
     xlim=c(start, end), ylim=c(start, end),
     cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(0, 1, col="orange", lwd=1.5, lty=1)
points(out$inflation_ave, out$inflation_hat_ave,,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
dev.off()

# plot: inflation vs. inflation_hat
start=min(c(dat$inflation,dat$inflation_hat))
end=max(c(dat$inflation,dat$inflation_hat))
    
png("fig/true-vs-predicted-inflation.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")
plot(dat$inflation, dat$inflation_hat,
     xlab="ave inflation", ylab="ave inflation hat",
     main=" bmi", 
     xlim=c(start, end), ylim=c(start, end),
     cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(0, 1, col="orange", lwd=1.5, lty=1)
points(dat$inflation, dat$inflation_hat,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
dev.off()


```

### good_chisq vs. adj_chisq

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/good_chisq-vs-adj_chisq.png"),
      width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # adjust chisq
  mod=lm(bad_chisq ~ r2_bad, data=use)
  use$adj_chisq=use$bad_chisq-mod$coefficients[2]*use$r2_bad
  
  # plot
  start=min(c(use$good_chisq,use$adj_chisq))
  end=max(c(use$good_chisq,use$adj_chisq))
  
  plot(use$good_chisq, use$adj_chisq,  
       xlab="good_chisq", ylab="bad_chisq - b_hat * ave_r2",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  points(use$good_chisq, use$adj_chisq, cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

dev.off()

```

### bad gwas: inf. vs. inf._hat: raw

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-vs-inflation-hat.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]

  # inflation  
  use$inflation=use$bad_chisq-use$good_chisq
  mod1=lm(bad_chisq~ldsc_ref, data=use)
  new=resid(mod1)+mod1$coefficients[1]-1
  mod2=lm(new~use$r2_bad)
  use$inflation_hat=predict(mod2)
  
  # plot: inflation vs. inflation_hat
  start=min(c(use$inflation,use$inflation_hat))
  end=max(c(use$inflation,use$inflation_hat))
  
  plot(use$inflation, use$inflation_hat,
       xlab="inflation", ylab="inflation hat",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(bad_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  points(use$inflation, use$inflation_hat,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

dev.off()


```

### bad gwas: inf. vs. inf._hat: ave

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-vs-inflation-hat-ave.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # inflation
  use$inflation=use$bad_chisq-use$good_chisq
  mod1=lm(bad_chisq~ldsc_ref, data=use)
  new=resid(mod1)+mod1$coefficients[1]-1
  mod2=lm(new~use$r2_bad)
  use$inflation_hat=predict(mod2)
  
  # bin a variable by quantile
  cutoff=quantile(use$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)
  use$bin=cut(use$r2_bad, breaks=cutoff, labels=1:(length(cutoff)-1))
    
  # average chisq by bin values
  out=data.frame(inflation_ave=tapply(use$inflation,INDEX=use$bin, mean),
                 inflation_hat_ave=tapply(use$inflation_hat,INDEX=use$bin, mean))
  out$bin_val=tapply(use$r2_bad,INDEX=use$bin, mean)

  # plot
  start=min(c(out$inflation_ave,out$inflation_hat_ave))
  end=max(c(out$inflation_ave,out$inflation_hat_ave))
  
  plot(out$inflation_ave, out$inflation_hat_ave,
       xlab="ave inflation", ylab="ave inflation hat",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(bad_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  points(out$inflation_ave, out$inflation_hat_ave,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

dev.off()

```

### bad gwas: inf. from full model

Here we estimate the inflation due to confounding from a full model. Basically, we added $\overline{r^2_j}$ as a predictor to the LD score regression to allow the inflation term to vary across SNPs by $\overline{r^2_j}$. Here is the model:

$S_j = \beta_0 + \beta_1 \overline{r^2_j} + \beta_2 l_j + \epsilon_j$

In this model, $\hat{\beta_2} l_j$ gives the estimated inflation due to polygenicity, and $\hat{\beta_0} + \hat{\beta_1} \overline{r^2_j} -1$ gives the estimated inflation due to confounding from population stratification. When a GWAS is free from confounding, $\hat{\beta_0} + \hat{\beta_1} \overline{r^2_j} -1$ should be very close to zero.

Using $\hat{\beta_0} + \hat{\beta_1} \overline{r^2_j} -1$ as an estimate of inflation makes sense. Here is the reasoning: 
$E(\delta_j) = E(S_j) - E(S'_j)$, where $S_j$ and $S'_j$ are chisq from bad and good gwas, respectively. Based on LDSC, $E(S'_j)= 1 + \beta'_1l$, and $E(S_j)$ is based on the proposed method. But in real situations, we do not have $S'_j$ to regress on $l$. However, we noted that $S'_j= \beta'_0+ \beta'_1 l + \epsilon'_j $ and $S_j= \beta_0+ \beta_1 l + \epsilon_j$ have similar intercept estimate (i.e., $\beta'_1$ and $\beta_1$ estimates are similar), and $\beta'_0$ is close to 1. So we can basically get $S'_j$ estimate by regressing $S_j$ on ldscore.

For traits where confounding is detected (i.e., all but snoring), the correlations between predicted inflation (from the model) and estimated inflation (chisq from bad gwas - chisq from good gwas) are moderate to high, ranging between 0.16 (pulse) to 0.93 (reaction). This indicates that the model is doing a good job of predicting inflation due to confounding when there is confounding. The predicted inflation given aver2 for each trait is summarized in the figure below. 

We also adjusted the chisq from bad gwas by taking out the estimated inflation. The cor(adjusted chisq, good chisq) is low in general. The correlation is essentially the correlation between residuals from the model $S_j = \beta_0 + \beta_1 \overline{r^2_j} + \beta_2 l_j + \epsilon_j$ and the residuals from the model $S'_j = \beta'_0 + \beta'_1 l_j + \epsilon'_j$ (where $S_j$ and $S'_j$ are chisq from bad and good GWASs, respectively). See table below.

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-vs-inflation-hat-full-mod.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # inflation
  use$inflation=use$bad_chisq-use$good_chisq
  mod=lm(bad_chisq~r2_bad + ldsc_ref, data=use)
  use$inflation_hat = mod$coefficients[1]+mod$coefficients[2]*use$r2_bad-1
  r=round(cor(use$inflation, use$inflation_hat),2)
  
  # plot inflation vs. inflation hat
  start=min(c(use$inflation,use$inflation_hat))
  end=max(c(use$inflation,use$inflation_hat))
  
  plot(use$inflation, use$inflation_hat,
       xlab="inflation (= bad-good chisq)", 
       ylab="inflation_hat (from the full model)",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(bad_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  text(start, end, paste0("r = ", r), adj=c(0,1), cex=2, font=2)

}

dev.off()

# ave inflation ~ aver2 --------------------------------------------------------

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-hat-by-aver2-bin.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5)

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # inflation
  use$inflation=use$bad_chisq-use$good_chisq
  mod=lm(bad_chisq~r2_bad + ldsc_ref, data=use)
  use$inflation_hat = mod$coefficients[1]+mod$coefficients[2]*use$r2_bad-1
  
  # bin a variable by quantile
  cutoff=quantile(use$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)
  use$bin=cut(use$r2_bad, breaks=cutoff, labels=1:(length(cutoff)-1))
    
  # average chisq by bin values
  out=data.frame(inflation_ave=tapply(use$inflation,INDEX=use$bin, mean),
                 inflation_hat_ave=tapply(use$inflation_hat,INDEX=use$bin, mean))
  out$bin_val=tapply(use$r2_bad,INDEX=use$bin, mean)

  # plot
  plot(out$bin_val, out$inflation_ave,
       ylab="mean inflation", xlab=" aver2 bin",
       main=trait, cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  lines(out$bin_val, out$inflation_hat_ave, 1, col="orange", lwd=1.5, type="l")
}

dev.off()

# extract relevant output----------------------------------------------------------------

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")


for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # inflation
  mod=lm(bad_chisq~r2_bad + ldsc_ref, data=use)
  use$chisq_adj = use$bad_chisq-(mod$coefficients[1]+mod$coefficients[2]*use$r2_bad-1)
  use$chisq_adj2= 1+(mod$coefficients[3]*use$ldsc_ref)
  
  r=round(cor(use$good_chisq, use$chisq_adj),3)
  use$p_bad=pchisq(use$bad_chisq,1, lower.tail=F)
  threshold=5*10^-8
  sel=use[use$p_bad<=threshold,] # significant gwas chisq only
  r2=round(cor(sel$good_chisq, sel$chisq_adj),3)
  
  # plot inflation vs. inflation hat
  start=min(c(use$good_chisq,use$chisq_adj))
  end=max(c(use$good_chisq,use$chisq_adj))
  
  plot(use$good_chisq,use$chisq_adj,
       xlab="chisq from good gwas", 
       ylab="adjusted chisq from bad gwas",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="gray", lwd=1.5, lty=1)
  text(start, end, paste0("r = ", r), adj=c(0,1), cex=2, font=2)
  points(sel$good_chisq,sel$chisq_adj,
         cex = 1.5, pch=21,  bg="orange", col="white", lwd=0.5)
  
  # collect output
  out0=data.frame(trait=trait,r_all=r, r_sig=r2)
  if(i==1){out=out0} else {out=rbind(out,out0)}
}

dev.off()

```


### To Explore: change??

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

#png(paste0("fig/inflation-vs-inflation-hat-ave-full-mod.png"),
#     width = 50, height = 30, units = "cm", res=600)
#par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  sel=c(1:2,5,6)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]
  
  # inflation
  use$inflation=use$bad_chisq-use$good_chisq
  mod0=lm(good_chisq~ldsc_ref, data=use)
  adj_chisq0=use$good_chisq-(mod0$coefficients[1]-1)
 
  # adjustment by ldsc
  mod1=lm(bad_chisq~ldsc_ref, data=use)
  mod2=lm(good_chisq~ldsc_ref, data=use)
  chisq1=1+mod1$coefficients[2]*use$ldsc
  chisq2=mod2$coefficients[1]+mod2$coefficients[2]*use$ldsc
  cor(chisq1,chisq2)
    
  adj_chisq1=1+mod1$coefficients[1]
  
  adj_chisq1=use$bad_chisq-(mod1$coefficients[1]-1)
  cor(use$good_chisq, adj_chisq1)
  
  # adjustment by new method
  mod2=lm(bad_chisq~r2_bad + ldsc_ref, data=use)
  adj_chisq2=use$bad_chisq-(mod2$coefficients[1]+mod2$coefficients[2]*use$r2_bad-1)
  cor(use$good_chisq, adj_chisq2)
  
  use$inflation=use$bad_chisq-use$good_chisq
  mod1=lm(use$inflation ~ poly(use$ldsc_ref,2) )
  mod2=lm(use$inflation ~ use$ldsc_ref + poly(use$r2_bad,2) + use$ldsc_ref * poly(use$r2_bad,2) + poly(use$maf,2)*use$r2_bad )
  mod3=lm(use$inflation ~ use$ldsc_ref + use$r2_bad + use$maf)
  
  mod5=lm(use$good_chisq ~ use$r2_bad)
  
  # quantiles
  out=data.frame(good=use$good_chisq,
                 bad=use$bad_chisq,
                 adj_ldsc=adj_chisq1,
                 adj_aver2=adj_chisq2)
  
  # false positives
  sel=use$bad_chisq>qchisq(1-5*10^-8,1,lower.tail=T) & use$good_chisq<qchisq(1-5*10^-8,1,lower.tail=T)
  out=out[sel,]
  cor(out$good, out$adj_aver2)
  cor(out$good, out$adj_ldsc)
  
  
  # bin a variable by quantile
  qq=data.frame(good=quantile(out$good, probs = seq(0, 1, 0.01), na.rm=T),
                bad=quantile(out$bad, probs = seq(0, 1, 0.01), na.rm=T),
                adj_ldsc=quantile(out$adj_ldsc, probs = seq(0, 1, 0.01), na.rm=T),
                adj_aver2=quantile(out$adj_aver2, probs = seq(0, 1, 0.01), na.rm=T))
  
  
  
  mod0_pred=mod0$coefficients[1]+mod0$coefficients[2]*use$ldsc_ref
  mod1_pred=mod1$coefficients[1]+mod1$coefficients[2]*use$ldsc_ref
  
  mod=lm(bad_chisq~r2_bad + ldsc_ref, data=use)
  new_chisq=mod$coefficients[1]+mod$coefficients[3]*use$ldsc_ref
  out=cbind(use$good_chisq, new_chisq)
  use$inflation_hat = mod$coefficients[1]+mod$coefficients[2]*use$r2_bad-1
  
  # bin a variable by quantile
  cutoff=quantile(use$r2_bad, probs = seq(0, 1, 0.005), na.rm=T)
  use$bin=cut(use$r2_bad, breaks=cutoff, labels=1:(length(cutoff)-1))
    
  # average chisq by bin values
  out=data.frame(inflation_ave=tapply(use$inflation,INDEX=use$bin, mean),
                 inflation_hat_ave=tapply(use$inflation_hat,INDEX=use$bin, mean))
  out$bin_val=tapply(use$r2_bad,INDEX=use$bin, mean)

  # plot
  start=min(c(out$inflation_ave,out$inflation_hat_ave))
  end=max(c(out$inflation_ave,out$inflation_hat_ave))
  
  plot(out$inflation_ave, out$inflation_hat_ave,
       xlab="ave inflation", ylab="ave inflation hat",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(bad_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  points(out$inflation_ave, out$inflation_hat_ave,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

#dev.off()

```

### good gwas: inf. vs. inf._hat: raw

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-vs-inflation-hat-goodgwas.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,4,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]

  # inflation  
  use$inflation=use$control_chisq-use$good_chisq
  mod1=lm(good_chisq~ldsc_ref, data=use)
  new=resid(mod1)+mod1$coefficients[1]-1
  mod2=lm(new~use$r2_good)
  use$inflation_hat=predict(mod2)
  
  # plot: inflation vs. inflation_hat
  start=min(c(use$inflation,use$inflation_hat))
  end=max(c(use$inflation,use$inflation_hat))
  
  plot(use$inflation, use$inflation_hat,
       xlab="inflation", ylab="inflation hat",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(good_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  points(use$inflation, use$inflation_hat,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

dev.off()

```

### good gwas: inf. vs. inf._hat: ave

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-vs-inflation-hat-ave-goodgwas.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5, pty="s")

for(i in 1:length(traits)){

  # putting info together
  trait=traits[i]
  file=paste0("summary/", trait, "-gwas-test-stats-compare2.txt")
  dat=vroom(file, col_names=T)
  r2=vroom(paste0("inflation/summary/",trait,"-inflation-by-maf-r2"), col_names=T)
  m=match(r2$snp, dat$SNP)
  use=data.frame(dat[m,c(1,4,5,6,2)], r2[,c(6:8)])
  use=use[complete.cases(use),]

  # inflation  
  use$inflation=use$control_chisq-use$good_chisq
  mod1=lm(good_chisq~ldsc_ref, data=use)
  new=resid(mod1)+mod1$coefficients[1]-1
  mod2=lm(new~use$r2_good)
  use$inflation_hat=predict(mod2)
  
 # bin a variable by quantile
  cutoff=quantile(use$r2_good, probs = seq(0, 1, 0.005), na.rm=T)
  use$bin=cut(use$r2_good, breaks=cutoff, labels=1:(length(cutoff)-1))
    
  # average chisq by bin values
  out=data.frame(inflation_ave=tapply(use$inflation,INDEX=use$bin, mean),
                 inflation_hat_ave=tapply(use$inflation_hat,INDEX=use$bin, mean))
  out$bin_val=tapply(use$r2_good,INDEX=use$bin, mean)

  # plot
  start=min(c(out$inflation_ave,out$inflation_hat_ave))
  end=max(c(out$inflation_ave,out$inflation_hat_ave))
  
  plot(out$inflation_ave, out$inflation_hat_ave,
       xlab="ave inflation", ylab="ave inflation hat",
       main=trait, 
       xlim=c(start, end), ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  abline(0, 1, col="orange", lwd=1.5, lty=1)
  abline(h=lm(good_chisq~ldsc_ref, data=use)$coefficients[1]-1, 
         col="red", lwd=1.5, lty=1)
  points(out$inflation_ave, out$inflation_hat_ave,cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
  
}

dev.off()

```

## UKBB N = 337k {#ukbb_337k_inflation}

From the above, we know that by assuming constant inflation, ldsc underestimates inflation. Then why for N = 337k, ldsc shows inflation in the absence of confouding?
Here we want to estimate inflation for the recommended 337k UKBB (inflation is estimated using the full model).

Note: previously we tested if the slope of the regression chisq ~ aver2 greater than 0 or not. Here we want to get the estimated inflation from the full model, i.e., chisq ~ aver2 + ldsc. So we need to reorganize information.

### organize data

```{bash eval=F}

#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# extract LD score
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# convert snp list to rs system ------------------------------------------------
infile=gen/snps-norm.use
outfile=gen/snps-norm.use-rs
awk '(NR==FNR){a[$1]; next}
     ($1 in a){print $1, $2}' $infile doug/ukbb.ldsc  > $outfile

#  extract ld scores -----------------------------------------------------------
dir=ldsc/eur_w_ld_chr
for chrom in {1..22}; do
zcat $dir/$chrom.l2.ldscore.gz | awk 'NR>1 {print $2, $6}' > ldscore
awk '(NR==FNR){a[$2]; b[$2]=$1; next}($1 in a){print b[$1], $2}' gen/snps-norm.use-rs ldscore > temp
if [ $chrom -eq 1 ]
then 
 mv temp snps-norm.ldscore
else
 cat snps-norm.ldscore temp > temp2
 mv temp2 snps-norm.ldscore
fi
echo $chrom
done

#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# putting information together
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

for trait in {awake,bmi,chron,ever,fvc,height,imp,neur,pulse,quals,reaction,sbp,snoring,hyper}; do

gwas=gwas-norm-337k/$trait-linear.summaries
aver2=summary/ave-r2-10k-snps-ukbb-norm
ldsc=snps-norm.ldscore
out=gwas-norm-337k-out/$trait.out

awk 'NR>1 {print $1, $5}' $gwas > tmp/gwas.tmp

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $ldsc tmp/gwas.tmp > tmp/ldsc.tmp

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $aver2 tmp/gwas.tmp > tmp/aver2.tmp

# put info together
paste tmp/gwas.tmp \
      tmp/aver2.tmp \
      tmp/ldsc.tmp \
      | awk 'BEGIN{OFS=";" ; 
                  print "snp;chisq;aver2;ldsc"}
             {$1=$1}1' > $out
done

```

### plot

```{r eval=F}

require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

png(paste0("fig/inflation-hat-aver2-bin-ukbb-recommended.png"),
     width = 50, height = 30, units = "cm", res=600)
par(mfrow=c(3,5), cex.lab=1.2, font.lab=2, cex.main=1.5)

for(i in 1:length(traits)){

  trait=traits[i]
  dat=vroom(paste0("gwas-norm-337k-out/",trait,".out"), col_names=T)
  dat=dat[complete.cases(dat),]
  
  # predicted inflation
  mod=lm(chisq~aver2 + ldsc, data=dat)
  mod1=lm(chisq~ ldsc, data=dat)
  mod2=lm(chisq~ aver2, data=dat)
  dat$inflation_hat = mod$coefficients[1]+mod$coefficients[2]*dat$aver2-1
  
  # bin a variable by quantile
  cutoff=quantile(dat$aver2, probs = seq(0, 1, 0.005), na.rm=T)
  dat$bin=cut(dat$aver2, breaks=cutoff, labels=1:(length(cutoff)-1))
    
  # average chisq by bin values
  out=data.frame(inflation_hat_ave=tapply(dat$inflation_hat,INDEX=dat$bin, mean))
  out$bin_val=tapply(dat$aver2,INDEX=dat$bin, mean)

  # plot
  start=min(out$inflation_hat_ave)
  end=max(out$inflation_hat_ave)
  
  plot(out$bin_val, out$inflation_hat_ave,
       xlab="ave r2 bin", ylab="ave inflation hat",
       main=trait,  ylim=c(start, end),
       cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
}

dev.off()

## Detection of inflation-------------------------------------------------------
require(vroom)
traits=c("awake","bmi","chron","ever",
        "neur","pulse","quals", "fvc", "height","imp", 
        "reaction","sbp","snoring","hyper")

for(i in 1:length(traits)){

  trait=traits[i]
  dat=vroom(paste0("gwas-norm-337k-out/",trait,".out"), col_names=T)
  dat=dat[complete.cases(dat),]
  
  mod=lm(chisq~aver2 + ldsc, data=dat)
  p0=data.frame(trait=trait,p_aver2=summary(mod)$coefficients[2,4])
  
  if(i==1){p=p0} else {p=rbind(p,p0)}
}


```


