
# previous GWAS

Here we conduct a GWAS that mimic the QC steps taken by some of the most inclusive GWAS studies. For example, the GWAS conducted by Codd et al. (2021) on telomere length has the following QC step, which resulted in n = 472,174 and m = 19.4 million SNPs

> We used imputed genotypes available in the UKB2 for the GWAS. To ensure quality, we restricted the analysis to variants with a MAF of ≥0.1% (where imputation accuracy is greatest) and an INFO score of ≥0.3. We tested 19.4 million variants using the BOLT-LMM package, adjusting for age, sex, array and the first ten principal components (PCs). The analysis was run separately for chromosome 23, where males were coded as 0/2.

Here we want to test if the test statistics from GWAS conducted in such a way are inflated using the proposed method. We do not have access to telomere length, but we can use the 14 traits as examples instead.

## QC

```{bash eval=F}

#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# extract covariates
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# This has been done under 'Good GWAS' section. Here we just want to extract from the file
R
options(scipen=999)
require(vroom)
dat=vroom("phen/covariates.phen", col_names=T)
pc=paste0("pc",1:10)
cov=c("eid","eid","age","geno_batch",pc)
out=dat[,cov]

write.table(out,"phen/covariates-inclusive-gwas.use", col.names=F,
            row.names=F, quote=F)
write.table(cov,"phen/covariates-inclusive-gwas.use-names", col.names=F,
            row.names=F, quote=F)

sel=out[complete.cases(out),] # n = 488,244
write.table(sel[,1:2], "complete-cases-covariates-inclusive-gwas.id", col.names=F,
            row.names=F, quote=F)

#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# id list of complete cases: covariates & phenotypes
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# note: to remain as much people as possible, we did not select all 14 traits. but 9.
# for these traits, n = 467, 104 have complete data
cp complete-cases-covariates-inclusive-gwas.id inclusive-gwas.id
dir=phen/continuous-traits/
# {awake,bmi,chron,ever,fvc,height,imp,neur,pulse,quals,reaction,sbp,snoring,hyper}
for tt in {awake,bmi,ever,height,imp,quals,reaction,imp,hyper}; do
mv inclusive-gwas.id temp
awk '(NR==FNR){a[$1];next}($1 in a){print $1, $2}' temp $dir/$tt.raw.pheno > inclusive-gwas.id
wc -l inclusive-gwas.id
echo $tt
done
rm temp

#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# extract genotypes
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# QC SNPs-----------------------------------------------------------------------
# stating number of SNPs=
for j in {1..22}; do
echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 20G
#SBATCH -c 3
#SBATCH -t 10:0:0

./plink2 --pfile ../gen/geno_plink/bhr$j \
         --keep ../inclusive-gwas.id \
         --hwe 0.0001 \
         --maf 0.001 \
         --hard-call-threshold .05 \
         --mach-r2-filter 0.3 2 \
         --make-bed \
         --memory 20000 \
         --out ../gen/tmp/bhr$j \
         --threads 3
"> sh_script/chr$j.sh
done

# submit jobs-------------------------------------------------------------------
for j in {1..22}; do
sbatch -A snpher ../sh_script/chr$j.sh
done > ../job-records/qc-inclusive-gwas

# check job completion----------------------------------------------------------
file=job-records/qc-inclusive-gwas
jobs=`awk '{print $4}' $file`
mkdir $file-tmp
for i in $jobs; do
jobinfo $i | awk -F ":" -v i=$i '$1~/Name/ {print i, $2}' >> $file-tmp/name.tmp
jobinfo $i | awk -F ":" '$1~/State/ {print$2}' >> $file-tmp/state.tmp
jobinfo $i | awk -F ":" '$1~/Cores/ {print$2}' >> $file-tmp/cores.tmp
jobinfo $i | awk -F ":" '$1~/Used walltime/ {print $2 ":" $3 ":" $4}' >> $file-tmp/time.tmp
jobinfo $i | awk -F ":" '$1~/Max Mem/ {split($2,a,/[(]/ ); print a[1]}' >> $file-tmp/mem.tmp
done
paste $file-tmp/name.tmp \
      $file-tmp/state.tmp \
      $file-tmp/cores.tmp \
      $file-tmp/time.tmp \
      $file-tmp/mem.tmp \
      | awk 'BEGIN{print "ID name state cores time mem"}{print $0}' > $file.out
rm -r $file-tmp

# merge files-------------------------------------------------------------------
rm gen/bfile-inclusive-gwas.list
for j in {1..22}; do
echo  "../gen/tmp/bhr$j" >> gen/bfile-inclusive-gwas.list
done

echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 80G
#SBATCH -c 10
#SBATCH -t 10:0:0
./ldak5.1 --make-bed ../gen/geno-inclusive-gwas \
          --mbfile ../gen/bfile-inclusive-gwas.list \
          --max-threads 10 \
          --exclude-dups YES
"> sh_script/mbfile-inclusive-gwas.sh

# submit the script-------------------------------------------------------------
sbatch -A snpher ../sh_script/mbfile-inclusive-gwas.sh >../job-records/mbfile-inclusive-gwas

# compute info score------------------------------------------------------------
for j in {1..22}; do
echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 20G
#SBATCH -c 3
#SBATCH -t 5:0:0

./plink2 --pfile ../gen/geno_plink/bhr$j \
         --threads 3 \
         --keep ../inclusive-gwas.id \
         --memory 20000 \
         --freq cols=chrom,ref,alt,altfreq,machr2 \
         --out ../gen/info/chr$j
"> sh_script/info-chr$j.sh
done

# submit file-------------------------------------------------------------------
for j in {1..22}; do
sbatch -A snpher ../sh_script/info-chr$j.sh
done > ../job-records/compute-info-score-inclusive-gwas

# combine files ----------------------------------------------------------------
dir=gen/info
rm $dir/inclusive-gwas-afreq
for i in {1..22}; do
if [ $i == 1 ]; then
  awk '{print $0}' $dir/chr$i.afreq > $dir/inclusive-gwas-afreq
else
  awk 'NR>1 {print $0}' $dir/chr$i.afreq >> $dir/inclusive-gwas-afreq
fi
done

#::::::::::::::::::
# SNP list
#::::::::::::::::::

# MAF & call-rate
awk < gen/bfile-inclusive-gwas.stat '($5>.001 && $6>=0.95 && NR>1){print $1}' > snps-inclusive-gwas.use
# m = SNPs
```

## gwas-telomere length

### aver2_j
Here we want to check for evidence of inflation in the telomere GWAS statistics (Codd et al. 2021) due to confouding from population stratification.

Here are the steps:

  1. download the GWAS stats (https://figshare.com/s/caa99dc0f76d62990195)
  2. randomly chose 10k SNPs from each side of the genome
  3. compute aver2_j for the 20k SNPs
  4. check if the chi square test stats vary with respect to aver2_j.

We show below that the slope of the regression line is not sig. different from zero. Hence, there is no evidence of confounding by the proposed method. It should be noted though that we used the genotype data of n = 487,409 individuals to compute aver2_j, and the original GWAS was based on n = 472,174.
  
```{bash eval=F}
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# select SNPs & organize data
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# download gwas stats-----------------------------------------------------------
mkdir gwas-telomere
wget https://figshare.com/ndownloader/files/28414941?private_link=caa99dc0f76d62990195
# m = 20,134,422 SNPs

# organize data: snp name & chisq ----------------------------------------------
infile=gwas-telomere/UKB_telomere_gwas_summarystats.tsv.gz
outfile=gwas-telomere/telemere-gwas-organized
zcat $infile | awk -F '\t' 'BEGIN {print "chr base_pair_location chr_base rs chisq"} 
                            NR >1 {print $3, $4,$3":"$4, $1, ($8/$9)^2}' > $outfile
                            
# hapmap3 SNPs------------------------------------------------------------------
infile1=doug/ukbb.ldsc
infile2=gwas-telomere/telemere-gwas-organized
outfile=gwas-telomere/telemere-gwas-organized-hm3
awk 'NR==FNR{a[$3];next} 
     ($3 in a || FNR==1 ){print $0}' $infile1 $infile2 > $outfile

# choose snp lists--------------------------------------------------------------
# choose 40k for each side as the pool for selection
m=40000
infile=gwas-telomere/telemere-gwas-organized-hm3
left=inflation/gwas-telomere/left-snps-pool
right=inflation/gwas-telomere/right-snps-pool
awk '$1<8 && NR > 1 {print $1, $2, $3}' $infile | shuf | head -n $m >$left
awk '$1>=8 && NR > 1 {print $1, $2, $3}' $infile | shuf | head -n $m >$right

# entire snp list of ukbb
out=inflation/gwas-telomere/ukbb-bhr-all-snps
rm $out
for i in {1..22};do
gen=gen/geno_plink/bhr$i.pvar
awk '{print $1, $2, $1":"$2, $3}' $gen >> $out
done
# wc -l gen/geno_plink/bhr{1..22}.pvar

# put the chosen 40k snps in the right format
snp=inflation/gwas-telomere/ukbb-bhr-all-snps
leftin=inflation/gwas-telomere/left-snps-pool
rightin=inflation/gwas-telomere/right-snps-pool
leftout=inflation/gwas-telomere/left-snps-tmp
rightout=inflation/gwas-telomere/right-snps-tmp

awk '(NR==FNR){a[$3];next}
     ($3 in a){print $4}' $leftin $snp > $leftout

awk '(NR==FNR){a[$3];next}
     ($3 in a){print $4}' $rightin $snp > $rightout
# note:  > 40k are in the list of each side because some snps have the same basepair...    

#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
#  extract genotypes of the bfiles
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

left=inflation/gwas-telomere/left-snps-tmp
right=inflation/gwas-telomere/right-snps-tmp
out=inflation/gwas-telomere/snp-all
awk '{print $0}' $left $right > $out

# extract data -----------------------------------------------------------------
for j in {1..22}; do
echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 4G
#SBATCH -c 3
#SBATCH -t 00:30:0

./plink2 --pfile ../gen/geno_plink/bhr$j \
         --extract ../inflation/gwas-telomere/snp-all \
         --make-bed \
         --out ../gen/tmp/bhr$j \
         --memory 20000 \
         --threads 3
"> sh_script/chr$j.sh
done

# submit jobs-------------------------------------------------------------------
for j in {1..22}; do
sbatch -A snpher ../sh_script/chr$j.sh
done > ../job-records/bfile-telomere

# check job completion----------------------------------------------------------
file=job-records/bfile-telomere
jobs=`awk '{print $4}' $file`
mkdir $file-tmp
for i in $jobs; do
jobinfo $i | awk -F ":" -v i=$i '$1~/Name/ {print i, $2}' >> $file-tmp/name.tmp 
jobinfo $i | awk -F ":" '$1~/State/ {print$2}' >> $file-tmp/state.tmp
jobinfo $i | awk -F ":" '$1~/Cores/ {print$2}' >> $file-tmp/cores.tmp
jobinfo $i | awk -F ":" '$1~/Used walltime/ {print $2 ":" $3 ":" $4}' >> $file-tmp/time.tmp
jobinfo $i | awk -F ":" '$1~/Max Mem/ {split($2,a,/[(]/ ); print a[1]}' >> $file-tmp/mem.tmp
done
paste $file-tmp/name.tmp \
      $file-tmp/state.tmp \
      $file-tmp/cores.tmp \
      $file-tmp/time.tmp \
      $file-tmp/mem.tmp \
      | awk 'BEGIN{print "ID name state cores time mem"}{print $0}' > $file.out
rm -r $file-tmp

# merge files-------------------------------------------------------------------
rm gen/bfile.list
for j in {1..22}; do
echo  "../gen/tmp/bhr$j" >> gen/bfile.list
done

echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 20G
#SBATCH -c 3
#SBATCH -t 01:00:0
./ldak5.1 --make-bed ../gen/geno-telomere \
          --mbfile ../gen/bfile.list \
          --max-threads 3 \
          --exclude-dups YES
"> sh_script/mbfile-telomere.sh

# submit the script-------------------------------------------------------------
sbatch -A snpher ../sh_script/mbfile-telomere.sh >../job-records/mbfile-telomere

# compute info score------------------------------------------------------------
for j in {1..22}; do
snp=../inflation/gwas-telomere/snp-all
out=../gen/info/chr$j-telomere
echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 5G
#SBATCH -c 3
#SBATCH -t 00:20:0

./plink2 --pfile ../gen/geno_plink/bhr$j \
         --threads 3 \
         --extract $snp \
         --memory 20000 \
         --freq cols=chrom,ref,alt,altfreq,machr2 \
         --out $out
"> sh_script/info-chr$j.sh
done

# submit file-------------------------------------------------------------------
for j in {1..22}; do
sbatch -A snpher ../sh_script/info-chr$j.sh
done > ../job-records/compute-info-telomere

# combine files ----------------------------------------------------------------
dir=gen/info
rm $dir/telomere-afreq
for i in {1..22}; do
if [ $i == 1 ]; then
  awk '{print $0}' $dir/chr$i-telomere.afreq > $dir/telomere-afreq
else
  awk 'NR>1 {print $0}' $dir/chr$i-telomere.afreq >> $dir/telomere-afreq
fi
done

#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
#  Finalize the 10k SNP list of each side
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

m=10000
snp=gen/geno-telomere.bim
leftin=inflation/gwas-telomere/left-snps-tmp
rightin=inflation/gwas-telomere/right-snps-tmp
leftout=inflation/gwas-telomere/left-snps
rightout=inflation/gwas-telomere/right-snps

awk '(NR==FNR){a[$1];next}
     ($2 in a){print $2}' $leftin $snp | shuf | head -$m  > $leftout

awk '(NR==FNR){a[$1];next}
     ($2 in a){print $2}' $rightin $snp | shuf | head -$m  > $rightout

#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
#  compute ave r^2_j
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

lista=../inflation/gwas-telomere/right-snps
listb=../inflation/gwas-telomere/left-snps
bfile=../gen/geno-telomere
out=../inflation/gwas-telomere/out/10k-snps

echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 80G
#SBATCH -c 7
#SBATCH -t 00:30:0
./ldak5.2 --max-threads 7 \
          --calc-inflation $out \
          --bfile $bfile \
          --lista $lista \
          --listb $listb
">sh_script/calc-r-10k-snps

# submit the job
sbatch -A snpher ../sh_script/calc-r-10k-snps >../job-records/calc-r-telomere

#  compute ave r_j^2 for each i-------------------------------------------------
R
require(vroom)

nm="10k-snps"
dat=vroom(paste0("inflation/gwas-telomere/out/",nm,".pairwise"), col_names=F)
lista=read.table(paste0("inflation/gwas-telomere/out/",nm,".predictorsa"), 
                stringsAsFactors = F)
listb=read.table(paste0("inflation/gwas-telomere/out/",nm,".predictorsb"), 
                stringsAsFactors = F)

dat=dat[,-c(10001)]^2
outb=data.frame(predictor=listb$V1, ave_r2=apply(dat,2, mean))
outa=data.frame(predictor=lista$V1, ave_r2=apply(dat,1, mean))
out=rbind(outa,outb)

write.table(out,paste0("summary/ave-r2-",nm, "-ukbb-telomere"), col.names=F, row.names=F, quote=F)  

```

### organize data

```{bash eval=F}
# NEED TO CHECK THE FOLLOWING CODE
aver2=summary/ave-r2-10k-snps-ukbb-telomere
rs=doug/ukbb.ldsc
gwas=gwas-telomere/telomere-gwas-organized-hm3
out=gwas-telomere/telomere.out

awk 'NR==FNR {a[$1]; b[$1]=$2; next}
      ($1 in a) {print $1, b[$1], $2}' $rs $aver2 > tmp/aver2.tmp
      
awk 'BEGIN{print "snp rs chisq aver2"} 
     NR==FNR {a[$2]; b[$2]=$1; c[$2]=$3; next}
     ($4 in a) {print b[$4], $4, $5, c[$4] }' tmp/aver2.tmp $gwas  > $out  

```

### chisq ~ aver2_j

```{r eval=F}
require(vroom)
dat=vroom("gwas-telomere/telomere.out", col_names=T)

# estimate slope: chisq ~ aver2_j-----------------------------------------------
mod=lm(chisq ~ aver2,data=dat)
summary(mod)

#Coefficients:
#              Estimate Std. Error t value Pr(>|t|)    
#(Intercept)    1.65142    0.09146  18.057   <2e-16 ***
#aver2       -196.61028  248.64693  -0.791    0.429    

# make the plot: binned aver2_j-------------------------------------------------

# bin a variable by quantile
cutoff=quantile(dat$aver2, probs = seq(0, 1, 0.005), na.rm=T)
dat$bin=cut(dat$aver2, breaks=cutoff, labels=1:(length(cutoff)-1))
# average chisq by bin values
out=data.frame(chisq_ave=tapply(dat$chisq,INDEX=dat$bin, mean))
out$bin_val=tapply(dat$aver2,INDEX=dat$bin, mean)

png("fig/chisq-by-aver2-bin-telomere.png",
    width = 10, height = 10, units = "cm", res=600)
start=0
end=max(out$chisq_ave)
plot(out$bin_val, out$chisq_ave,
         xlab="ave r2", ylab="ave chisq",
         ylim=c(start,8),
         main="", las=1,
         cex = 0.8, pch=21,  bg="orange", col="white", lwd=0.5)
dev.off()


# make the plot: raw aver2_j & chisq--------------------------------------------

png("fig/chisq-by-aver2-bin-telomere-raw.png",
    width = 10, height = 10, units = "cm", res=600)
start=0
end=max(dat$chisq)
plot(dat$aver2, dat$chisq,
         xlab="ave r2", ylab="chisq",
         ylim=c(start,400),
         main="", las=1,
         cex = 0.8, pch=21,  bg="orange", col="white", lwd=0.5)
dev.off()

```

## PR interval GWAS

Here we want to check whether previous GWAS was affected by confounding. 
We use the PR interval GWAS as an example to work out the workflow.

### aver2_j

```{r eval=F}
mkdir previous-gwas-sumstats

# common SNPs: 1kg, gwas & ldsc-------------------------------------------------

# 1kg & ldsc
dir=previous-gwas-sumstats
ldsc=$dir/snp-list/ldsc.snps
snp1kg=$dir/snp-list/1kg.snps
out=$dir/snp-list/ldsc-1kg-common.snps

awk '{a=$5; 
      b=$6;
      if(a<b) 
      print $2, $2 "_" a "_" b;
      else 
      print $2, $2 "_" b "_"a}' \
      $snp1kg > 1kg.tmp

awk '{a=$3; 
      b=$4;
      if(a<b) 
      print $1, $2"_" a "_" b;
      else 
      print $1, $2"_" b "_" a}' \
      $ldsc > ldsc.tmp

awk 'NR==FNR {a[$2]; b[$2]=$1; next} 
     {if ($2 in a) print $1, b[$2], $2}' 1kg.tmp ldsc.tmp > $out

# gwas snps
dir=previous-gwas-sumstats
infile=$dir/PR_1000g_GWAS_ALL.zip
outfile=$dir/snp-list/pr.snps

zcat $infile | awk 'NR >1 {a=$2;
                 b=$3;
                 if(a<b)
                 print $1, $5 ":" $6 "_" toupper(a) "_" toupper(b);
                 else 
                 print $1, $5 ":" $6 "_" toupper(b) "_" toupper(a)}' > $outfile 

# select gwas snps common to 1kg & ldsc
dir=previous-gwas-sumstats
snp=$dir/snp-list/ldsc-1kg-common.snps
gwas=$dir/snp-list/pr.snps
outfile=$dir/pr-sel.snps

awk 'NR==FNR {a[$3]; next} 
     {if ($2 in a) print $0}' $snp $gwas > $outfile 

# organize gwas data------------------------------------------------------------
infile=previous-gwas-sumstats/PR_1000g_GWAS_ALL.zip
outfile=previous-gwas-sumstats/pr.gwas
zcat $infile | awk 'BEGIN {print "snp chr pos chisq p_val"}
                   NR >1 {print $1, $5, $6, ($7/$8)^2, $9}' > $outfile

# select gwas data for snps common with 1kg & ldsc
infile=previous-gwas-sumstats/pr.gwas
snpfile=previous-gwas-sumstats/pr-sel.snps
outfile=previous-gwas-sumstats/pr-sel.gwas
 
awk 'NR==FNR {a[$1]; b[$1]=$2 ; next}
     BEGIN {print "snp rs chr pos chisq p_val"}
     {if ($1 in a) print b[$1], $0}' $snpfile $infile > $outfile

# select 10k SNPs to compute ave r2_j-------------------------------------------
# selection based on percentiles of chisq
R
require(vroom)
dat=vroom("previous-gwas-sumstats/pr-sel.gwas", col_names=T)

# bin snps by percentiles of chisq to get a good range of test statistics
cutoff=quantile(dat$chisq, probs = seq(0, 1, 0.01), na.rm=T)
dat$chisq_bin=cut(dat$chisq, breaks=cutoff, labels=1:(length(cutoff)-1))

# sample 200 SNPs from each percentile bin
out=unlist(tapply(dat$snp, dat$chisq_bin, 
           function(x) sample(x, 200, replace=F)))

# check which chromosomes they come from
m=match(out,dat$snp)
sel=dat[m,]

write.table(sel[,1:2],"previous-gwas-sumstats/pr-20k.snps",
            col.names=F, row.names=F, quote=F)

# compute aver2 using 1kg genotype data-----------------------------------------
mkdir previous-gwas-sumstats/aver2

snp=previous-gwas-sumstats/snp-list/ldsc-1kg-common.snps
infile=previous-gwas-sumstats/pr-20k.snps
outfile=previous-gwas-sumstats/aver2/pr-20k.use

awk 'NR==FNR {a[$1]; next}
     {if ($3 in a) print $2, $2}' $infile $snp > $outfile

# calc aver2
list=../previous-gwas-sumstats/aver2/pr-20k.use
bfile=../ref-1kg/1kg/gen/1000b
out=../previous-gwas-sumstats/aver2/pr.aver2

echo "#"'!'"/bin/bash
#SBATCH --constraint \"s04|s05\"
#SBATCH --partition normal
#SBATCH --mem 30G
#SBATCH -c 7
#SBATCH -t 00:40:0
./ldak5.2 --max-threads 7 \
          --calc-inflation $out \
          --bfile $bfile \
          --lista $list \
          --listb $list
">sh_script/calc-aver2-pr

sbatch -A snpher ../sh_script/calc-aver2-pr

```

### organize data

```{bash eval=F}

# extract ld scores for SNPs common to 1kg & ldsc-------------------------------
dir=ldsc/eur_w_ld_chr
snps=previous-gwas-sumstats/snp-list/ldsc-1kg-common.snps
out=previous-gwas-sumstats/ldsc-1kg-common-snps.ldscore
for chrom in {1..22}; do
zcat $dir/$chrom.l2.ldscore.gz | awk 'NR>1 {print $2, $6}' > ldscore
awk '(NR==FNR){a[$1]; b[$1]=$2; c[$1]=$3; next}
     ($1 in a){print $1, b[$1], c[$1], $2}' $snps ldscore > temp
if [ $chrom -eq 1 ]
then 
 mv temp $out
else
 cat $out temp > temp2
 mv temp2 $out
fi
echo $chrom
done

# organize data ----------------------------------------------------------------
dir=previous-gwas-sumstats
gwas=$dir/pr-sel.gwas
snps=previous-gwas-sumstats/snp-list/ldsc-1kg-common.snps
aver2=$dir/aver2/pr.aver2.rjk2.average
ldsc=$dir/ldsc-1kg-common-snps.ldscore
out=$dir/pr.out

awk 'NR>1 {print $1, $5}' $gwas > tmp/gwas.tmp

awk 'NR==FNR {a[$3]; b[$3]=$4; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' $ldsc tmp/gwas.tmp > tmp/ldsc.tmp

awk 'NR==FNR {a[$2]; b[$2]=$3; next} 
     {if ($1 in a) print b[$1], $2}' $snps $aver2 > tmp/aver2.tmp0

awk 'NR==FNR {a[$1]; b[$1]=$2; next} 
     {if ($1 in a) print b[$1]; 
     else print "NA" }' tmp/aver2.tmp0 tmp/gwas.tmp > tmp/aver2.tmp

# put info together
paste tmp/gwas.tmp \
      tmp/aver2.tmp \
      tmp/ldsc.tmp \
      | awk 'BEGIN{OFS=";" ; 
                  print "snp;chisq;aver2;ldsc"}
             {$1=$1}1' > $out
```

### chisq ~ aver2_j

```{r eval=F}
require(vroom)

trait="pr"
dat=vroom(paste0("previous-gwas-sumstats/",trait,".out"), col_names=T)
dat=dat[complete.cases(dat),]

# linear model
mod=lm(chisq ~ aver2 + ldsc,data=dat)
p=summary(mod)$coefficients[,4][2]
p_sci=formatC(p, format="e", digit=2)

# bin a variable by quantile
cutoff=quantile(dat$aver2, probs = seq(0, 1, 0.005), na.rm=T)
dat$bin=cut(dat$aver2, breaks=cutoff, labels=1:(length(cutoff)-1))

# average chisq by bin values
out=data.frame(chisq_ave=tapply(dat$chisq,INDEX=dat$bin, mean))
out$bin_val=tapply(dat$aver2,INDEX=dat$bin, mean)
px=min(out$bin_val)
py=max(out$chisq_ave)

# plot binned chisq
png("fig/chisq-aver2-bin-pr-interval.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5)

plot(out$bin_val, out$chisq_ave,
           xlab="ave r2 bin", ylab="mean chisq",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(lm(chisq ~ aver2, data=dat),col="orange", lwd=2)
if(p<0.01){text(px, py, paste0("p = ", p_sci),
                adj=c(0,1), col="red", cex=2, font=2)
} else {text(px, py, paste0("p = ", p_sci), 
        adj=c(0,1), cex=2, font=2)}

dev.off()

# plot raw chisq
png("fig/chisq-aver2-pr-interval.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5)

px=min(dat$aver2)
py=max(dat$chisq)
plot(dat$aver2, dat$chisq,
           xlab="ave r2", ylab="chisq",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(lm(chisq ~ aver2, data=dat),col="orange", lwd=2)
if(p<0.01){text(px, py, paste0("p = ", p_sci),
                adj=c(0,1), col="red", cex=2, font=2)
} else {text(px, py, paste0("p = ", p_sci), 
        adj=c(0,1), cex=2, font=2)}

dev.off()

# test------------------

# bin a variable by quantile
cutoff=quantile(dat$chisq, probs = seq(0, 1, 0.01), na.rm=T)
dat$chisq_bin=cut(dat$chisq, breaks=cutoff, labels=1:(length(cutoff)-1))

# average chisq by bin values
out=data.frame(aver2_ave=tapply(dat$aver2,INDEX=dat$chisq_bin, mean))
out$chisq_bin_val=tapply(dat$chisq,INDEX=dat$chisq_bin, mean)

png("fig/chisq-aver2-pr-interval-test.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5)

plot(out$chisq_bin_val[60:99], out$aver2_ave[60:99],
           xlab="chisq", ylab="ave r2",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)
abline(lm(chisq_bin_val[80:99] ~ aver2_ave[80:99], data=out),col="orange", lwd=2)
dev.off()


png("fig/chisq-aver2-pr-interval-test2.png", res=600, width=30, height=20, units="cm")
par(cex.lab=1.2, font.lab=2, cex.main=1.5)

plot(1:99, out$chisq_bin_val[1:99],
           xlab="chisq", ylab="ave r2",
           main=trait, las=1,
           cex = 1.5, pch=21,  bg="darkgray", col="white", lwd=0.5)

dev.off()

```

## Ben Neale Pan-UKBB

```{bash eval=F}
mkdir neale-pangwas

dir=neale-pangwas
infile=$dir/biomarkers-30600-both_sexes-irnt.tsv.bgz
outfile=$dir/biomarkers-30600.gwas

zcat $infile | awk 'BEGIN{print "snp chisq"}
                     NR>1{a=$3;b=$4;
                     if(a<b && $6!="NA" && $7!="NA")
                     print $1":"$2"_"$3"_"$4, ($6/$7)^2;
                     if(a>b && $6!="NA" && $7!="NA")
                     print $1":"$2"_"$4"_"$3,($6/$7)^2}' > $outfile
                     


```



## notes/concerns

 * Telomere length GWAS:
    * The QC seems too rough so that some SNPs have multiple alleles.  



