---
title: "Quality control"
author: " "
date: "updated on `r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib, paper.bib]
biblio-style:
csl: apa.csl
link-citations: yes
github-repo: rstudio/bookdown-demo
description: " "
---

# General Info

## Topic

We propose a new method to detect the inflation of GWAS test statistics caused by population stratification (& cryptic relatedness). Here we document all analyses and results.

**Background**:

1. Existing methods: use summary statistics to detect and correct inflation in GWAS test statistics.

  * genomic inflation factor:  
    * $S_{median}$/$\chi^2 (1)_{median}$ > 1 ? 
    * does not account for inflation due to polygenicity
    
  * LD score regression [@bulik2015ld] (LDSC):
    * $S_{j} = 1 + n_{j}a +  (n_{j}\sum_{i}^{k}r_{ji}^2) (q_{j}/Q) h_{snp} + \epsilon_{j}$
      * under GCTA: $q_{j}=1, Q=\sum_{j}^{m}q_j=m$ 
      * under LDAK-thin: $q_{j}=I_{j}[f_{j}(1-f_{j})]^{0.75}$
    * It separates the inflation due to confounding (intercept) and the inflation due to polygenicity (slope).
    * The inflation due to confounding may not be constant but SNP specific instead [@holmes]. Hence, summary statistics may not be good after all to detect the inflation due to confounding. 

2. Proposed Method: use individual level data to detect inflation and maybe extended to summary statistics (?).
    
  * $T = (\hat{h^2}_{right} + \hat{h^2}_{left}) - \hat{h^2}_{whole} > 0$ ?
  * Confounding causes cross-chromosome correlations, such that each part of the genome tags each other. Hence, for confounded GWASs, $T = (\hat{h^2}_{right} + \hat{h^2}_{left}) - \hat{h^2}_{whole} > 0$.
  * when $h_{right}^2, h_{left}^2, h_{whole}^2$ are estimated using Haseman Elston regression, this test accounts to testing $tr(K_{left}K_{right}) = 0$ or average of SNP correlation^2 = 0.  
  

**Design**:

  * good GWASs: 100k unrelated white British
  * bad or confounding GWASs: 93k unrelated white British + 7k blacks & Asians 
  * control GWASs: 93k unrelated white British + 7k unrelated white British (who are not included in the good GWASs)
  * We also checked if confouding is evidence for the UKBB recommended white British (a randomly selection of  100k out of 337k), on which most UKBB GWASs are based on. 
  
**Results**:

So far the proposed and existing methods perform equally well.

  * good GWAS: no sig. inflation due to confounding
  * bad GWAS: sig. inflation due to confounding
  * control GWAS: no sig. inflation due to confounding
  * UKBB recommended: no evidence of inflation by any of the methods.

The proposed method does not seem to be more advantageous than existing methods. However, this does not mean that the existing methods are without problems. The inflation may not be constant as assumed by existing methods.

## Frequently used commands

`ssh -l zhoux login.genome.au.dk`

`sftp zhoux@login.genome.au.dk`

`lcd /home/zhoux/Dropbox/github/quality-control/main-files`

`cd /home/zhoux/dsmwpred/xuan/quality-control/qc-10oct`

`srun --mem=15g -c 2 -t 5:0:0 --constraint "s04|s05" -A snpher --pty /bin/bash`



